src/server/ua_services_subscription.c

```diff
@@ -395,7 +429,7 @@ Operation_SetMonitoringMode(UA_Server *server, UA_Session *session,
         return;
     }
 
-    if(mon->monitoredItemType != UA_MONITOREDITEMTYPE_CHANGENOTIFY) {
+    if(mon->monitoredItemType == UA_MONITOREDITEMTYPE_STATUSNOTIFY) {
```

Comment:
We should have a catch-all for unsupported types.
A user could send non-existing enums. Since this is transported as just an int32.